### PR TITLE
v2.0

### PR DESCRIPTION
**Pull request by liam-bh**
Improved latency of content being hidden and reworked how script is called. 

Huge stability improvements - unintended site components breaking occurs notably less. 

The stability improvements are likely due to using a CSS approach instead. Components can still interact with each other as intended, they are just hidden.

Overall simplified.

If you have a moment to look at it, I would greatly appreciate feedback.

If you have nothing to comment thats OK to.

### CODE REVIEW

File: injection.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+const customCSS = ".ytd-rich-grid-renderer, .ytd-item-section-renderer, .ytd-compact-video-renderer { opacity: 0; pointer-events: none; }";
```
Comment (liam-bh) [inline injection.js:2]: `.ytd-item-section-renderer` removes the comments section as well and will need to be changed.

Though considering the point of this extension is to reduce unproductive time on youtube it's arguably a good thing.

Comment (SleepySwords) [reply injection.js:2]: This could be a togglable setting. And you could have a pop-up settings menu, like other extensions.

File: injection.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	styleElement.parentNode.removeChild(styleElement);
+}
+
+// check if current url is allowed, and exectute accordingly
+function URLchecker() {
+	if(allowedPaths.includes(window.location.pathname)) {
+		if(!document.getElementById("injected-css")) {
+			addStyle();
+		}
+	} else {
```
Comment (SleepySwords) [inline injection.js]: Another nitpick, but you can combine this with the below if statement to form an `else if`

File: injection.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+// remove styling
+function removeStyle() {
+	let styleElement = document.getElementById("injected-css");
+	
+	styleElement.parentNode.removeChild(styleElement);
+}
+
+// check if current url is allowed, and exectute accordingly
+function URLchecker() {
```
Comment (SleepySwords) [inline injection.js]: Nitpick
The name of this function should be something like `updateStyle` or `checkAndUpdateStyle`.
I think it just gives just that tiny bit more information.

**Review by SleepySwords — DISMISSED** [review]
LGTM! The CSS approach is brilliant. Just two nitpicks.

**Comment by SleepySwords** [comment]
Oh, Brave just crashes when I tried to use it on Mac OS.

File: injection.js (hunk 1 above)
Comment (liam-bh) [reply injection.js:2]: Yeah if I feel like it actually becomes a problem then I'll add that in. V2.1 even.

**Comment by liam-bh** [comment]
I had huge issues with my browser crashing and then being unable to open when it errored, which was to do with issues in error logging. Chrome worked fine while Vivaldi did not.

The whole browser crashing shouldn't occur no matter what - errors should just be logged in chrome://extentions

So weird things going on I guess it isn't as stable as I believed.

Are you able to retrieve the error from Brave?

**Comment by SleepySwords** [comment]
Yeah, it's definitely weird, it works on Chrome for me as well.

I'll try to get those errors.

**Comment by SleepySwords** [comment]
Weirdly enough the errors I get is this
```
[1015/130753.828913:WARNING:process_memory_mac.cc(93)] mach_vm_read(0x7ffeef214000, 0x2000): (os/kern) invalid address (1)
[1015/130754.846829:WARNING:crash_report_exception_handler.cc(240)] UniversalExceptionRaise: (os/kern) failure (5)
fish: Job 1, '/Applications/Brave\ Browser.ap…' terminated by signal SIGILL (Illegal instruction)
```

It definitely looks like it's a Chromium/Brave issue.

**Review by SleepySwords — APPROVED** [review]

**Comment by liam-bh** [comment]
Yeah for sure.

**Pull request merged**